addons/account_peppol_selfbilling/models/account_edi_proxy_user.py

```diff
@@ -0,0 +1,36 @@
+from lxml import etree
+
+from odoo import models
+
+
+class AccountEdiProxyClientUser(models.Model):
+    _inherit = 'account_edi_proxy_client.user'
+
+    def _peppol_get_import_journal_and_move_type(self, attachment):
+        # Self-billed invoices are invoices which your customer creates on your behalf and sends you via Peppol.
+        # In this case, the invoice needs to be created as an out_invoice in a sale journal.
+        xml_tree = etree.fromstring(attachment.raw)
+        is_self_billed = False
+
+        # 329/527: Self-billing invoice; 261: Self-billing credit note
+        if xml_tree.findtext('.//{*}InvoiceTypeCode') in ['389', '527'] or xml_tree.findtext('.//{*}CreditNoteTypeCode') == '261':
+            is_self_billed = True
+
+        if not is_self_billed:
```

Comment:
is the variable needed here? Can't you just do all this in the if of the findtext?